account/account_test.go

```diff
@@ -530,6 +532,107 @@ func TestTransactionHashDeclare(t *testing.T) {
 	require.Equal(t, expectedHash.String(), hash.String(), "TransactionHashDeclare not what expected")
 }
 
+func TestWaitForTransactionReceiptMOCK(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+	mockRpcProvider := mocks.NewMockRpcProvider(mockCtrl)
+
+	mockRpcProvider.EXPECT().ChainID(context.Background()).Return("SN_GOERLI", nil)
+	acnt, err := account.NewAccount(mockRpcProvider, &felt.Zero, "", starknetgo.NewMemKeystore())
+	require.NoError(t, err, "error returned from account.NewAccount()")
+
+	type testSetType struct {
+		Timeout                      int
+		ShouldCallTransactionReceipt bool
+		Hash                         *felt.Felt
+		ExpectedErr                  error
+		ExpectedReceipt              rpc.TransactionReceipt
+	}
+	testSet := map[string][]testSetType{
+		"mock": {
+			{
+				Timeout:                      1000,
+				ShouldCallTransactionReceipt: true,
+				Hash:                         new(felt.Felt).SetUint64(1),
+				ExpectedReceipt:              nil,
+				ExpectedErr:                  errors.New("UnExpectedErr"),
+			},
+			{
+				Timeout:                      1000,
+				Hash:                         new(felt.Felt).SetUint64(2),
+				ShouldCallTransactionReceipt: true,
+				ExpectedReceipt: rpc.InvokeTransactionReceipt{
+					TransactionHash: new(felt.Felt).SetUint64(2),
+					ExecutionStatus: rpc.TxnExecutionStatusSUCCEEDED,
+				},
+				ExpectedErr: nil,
+			},
+			{
+				Timeout:                      1,
+				Hash:                         new(felt.Felt).SetUint64(3),
+				ShouldCallTransactionReceipt: false,
+				ExpectedReceipt:              nil,
+				ExpectedErr:                  context.DeadlineExceeded,
+			},
+		},
+	}[testEnv]
+
+	for _, test := range testSet {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(test.Timeout)*time.Second)
```

Comment:
small enhancement proposal: to change type of timeout field to time.Duration (there would be no need to cast integer constants to time.Duration i.e. `Timeout: time.Duration(1)` because constants in Go untyped and autocasts to expected type if conversion is possible)